lwt/nbd_lwt_unix.ml

```diff
@@ -50,16 +46,13 @@ let tls_channel_of_fd fd role () =
       | TlsServer ctx -> ctx, Lwt_ssl.ssl_accept
   in
   ssl_start fd ctx >>= fun sock ->
-  let offset = ref 0L in
 
   let read_tls buf =
-    io_complete "read" (Some !offset) Lwt_ssl.read_bytes sock buf >>= fun () ->
-    offset := Int64.(add !offset (of_int (Cstruct.len buf)));
+    io_complete Lwt_ssl.read_bytes sock buf >>= fun () ->
```

Comment:
Just a note for myself to better understand why `Lwt_ssl.read_bytes` and `Cstruct` are compatible:
[Lwt_types.t](https://ocsigen.org/lwt/2.6.0/api/Lwt_bytes) = [Cstruct.t.buffer](https://github.com/mirage/ocaml-cstruct/blob/master/lib/cstruct.mli)